sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileScanBuilder.scala

```diff
@@ -57,6 +64,27 @@ abstract class FileScanBuilder(
     StructType(fields)
   }
 
+  def pushFilters(partitionFilters: Seq[Expression], dataFilters: Seq[Expression]): Unit = {
+    this.partitionFilters = partitionFilters
+    this.dataFilters = dataFilters
+    this.pushedDataFilters = pushDataFilters(translateDataFilter)
+  }
+
+  protected def pushDataFilters(dataFilters: Array[Filter]): Array[Filter] = Array.empty[Filter]
+
+  protected def translateDataFilter(): Array[Filter] = {
```

Comment:
after https://github.com/apache/spark/pull/33650/files#r691835881 , we can inline this method?

Reply:
Done. Thanks!